File: app/src/main/java/com/nikola/jakshic/dagger/DaggerApp.java
package com.nikola.jakshic.dagger;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.nikola.jakshic.dagger.di.AppComponent;
import com.nikola.jakshic.dagger.di.NetworkModule;
import com.nikola.jakshic.dagger.di.DaggerAppComponent;

import io.fabric.sdk.android.Fabric;

public class DaggerApp extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        appComponent = DaggerAppComponent.builder()
                .networkModule(new NetworkModule(this))
                .build();
        Fabric.with(this, new Crashlytics());
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/com/nikola/jakshic/dagger/DaggerApp.kt
package com.nikola.jakshic.dagger

import android.app.Application
import androidx.hilt.work.HiltWorkerFactory
import androidx.work.Configuration
import coil.Coil
import coil.ImageLoader
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber
import timber.log.Timber.DebugTree
import javax.inject.Inject

@HiltAndroidApp
class DaggerApp : Application(), Configuration.Provider {
    @Inject lateinit var imageLoader: ImageLoader
    @Inject lateinit var workerFactory: HiltWorkerFactory

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }
        Coil.setImageLoader(imageLoader)
    }

    override fun getWorkManagerConfiguration(): Configuration {
        return Configuration.Builder()
            .setWorkerFactory(workerFactory)
            .build()
    }
}
